src/EVM.hs

```diff
@@ -1101,7 +1100,9 @@ callChecks this xGas xContext xTo xValue xInOffset xInSize xOutOffset xOutSize x
     accessMemoryRange xOutOffset xOutSize $ do
       availableGas <- use (#state % #gas)
       let recipientExists = accountExists xContext vm
-      let from = fromMaybe vm.state.contract vm.config.overrideCaller
+      let from = fromMaybe vm.state.contract vm.state.overrideCaller
+      resetCaller <- use $ #state % #resetCaller
+      when resetCaller $ assign (#state % #overrideCaller) Nothing
```

Comment:
This should probably also set resetCaller to False at the same time

Reply:
having this on callChecks feels a bit odd to me, should it maybe live on delegateCall instead?

Reply:
Hmmm... yeah, we could set `resetCaller` in `delegateCall`. However, the `let from = ..` needs to be here of course. Wanna have a go at putting it in `delegateCall`? Just create a PR on top of this PR :)

Reply:
Thanks, good point! I fixed it :) Let me know what you think!